### PR TITLE
Add PDK to idesdk section

### LINKED ISSUE #14: Add PDK to development (new section?)
[Puppet Development Toolkit](https://github.com/puppetlabs/pdk) (PDK) is only available via [manual download](https://puppet.com/download-puppet-development-kit) and [installation](https://puppet.com/docs/pdk/1.0/pdk_install.html).

Proposed Solution
---------------
A Puppet-based installation would look like this:

```puppet
  archive { 'pdk.deb':
    source  => 'https://pm.puppetlabs.com/cgi-bin/pdk_download.cgi?dist=ubuntu&rel=16.04&arch=amd64&ver=latest',
    path    => '/tmp/pdk.deb',
    extract => false,
    cleanup => false,
  }

  package { 'pdk':
    ensure   => present,  # latest? (to make sure dpkg upgrades an installed package)
    source   => '/tmp/pdk.deb',
    provider => 'dpkg',
  }
```

Maybe Reconsider `vcsscm`
-------------------
It may be worth reconsidering the modules structure for `vcsscm` (try to understand whether moving that under a `development` category would be clearer).

### PR DESCRIPTION
**Pull request by bittner**
These changes add PDK, the [Puppet Development Kit](https://puppet.com/download-puppet-development-kit), to the idesdk section. This is meant to live in this repo as long as Puppetlab's packages are not yet installable off repository servers (using the Puppet `package` resource).

Fixes #14.

### CODE REVIEW

**Comment by bittner** [comment]
We have the same two issues that make the build fail as in PR #16. This will go away with merging PR #15, I'm pretty sure.

**Comment by bittner** [comment]
:bell: **ping!**

This still seem valid even though Puppet Labs state [on the download page](https://puppet.com/download-puppet-development-kit) that the PDK is "available via native package managers". Seems a bit tricky though [picking the right Puppet version](https://puppet.com/docs/pdk/1.x/pdk_install.html#ariaid-title4) for the repository-based installation.

File: manifests/idesdk/pdk.pp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  case $::operatingsystem {
+    'Ubuntu': {
+      package { 'pdk':
+        ensure   => $ensure,
+        source   => '/tmp/pdk.deb',
+        provider => 'dpkg',
+      }
+
+      file { '/tmp/pdk.deb':
```
Comment (edestecd) [inline manifests/idesdk/pdk.pp]: I don't think we need to download the package for linux any longer.
If you are running puppet on a machine there is a good chance you installed it from the official repos, which also have pdk now. I think we can just get away with the package.

Comment (bittner) [reply manifests/idesdk/pdk.pp]: Maybe yes. For a pure PDK installation it's still more that just `package {'pdk':}`, though, unfortunately. That's still my motivation. You can look at my [Ansible-driven implementation](https://github.com/painless-software/ansible-software/blob/master/roles/developer/tasks/automation.yml#L4-L29) for comparison. It's 3 steps that you won't recall from the top of your head. They make sense to automate.

Comment (edestecd) [reply manifests/idesdk/pdk.pp]: Can it be optional to download the package?  I will use this and I can make the assumption that the puppet repo is setup on my machines. Can you use a param and do either?

File: manifests/idesdk/pdk.pp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# pdk.pp
+# Install Puppet Development Kit
+# https://puppet.com/download-puppet-development-kit
+#
+
+class software::idesdk::pdk (
+  $ensure  = $software::params::software_ensure,
+  $version = $software::params::pdk_version,
```
Comment (edestecd) [inline manifests/idesdk/pdk.pp]: Is $version used?

File: manifests/idesdk/pdk.pp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  $url     = $software::params::pdk_url,
+) inherits software::params {
+
+  validate_string($ensure)
```
Comment (edestecd) [inline manifests/idesdk/pdk.pp]: Can we get rid of the legacy validate. I think I have just been removing this one from the other profiles.

**Comment by edestecd** [comment]
resolved by bc856b57ab40240ad0d313e27d472c507f6e77d9

**Pull request closed without merging**